### PR TITLE
tree2: Document beta APIs

### PR DESCRIPTION
**Pull request by CraigMacomber**
## Description

Document all `@beta` APIs API-Extractor reported as undocumented.

This fixed all non-parameter property undocumented warnings for beta APIs. Parameter properties are currently incorrectly reporting as undocumented, as noted in https://github.com/microsoft/rushstack/issues/3462#issuecomment-1216986905

## Reviewer Guidance

The review process is outlined on [this wiki page](https://github.com/microsoft/FluidFramework/wiki/PR-Guidelines#guidelines).

### CODE REVIEW

File: experimental/dds/tree2/src/class-tree/schemaTypes.ts [hunk 1]
```diff
@@ -131,9 +148,23 @@ export enum FieldKind {
  * @beta
  */
 export enum NodeKind {
+	/**
+	 * Node which serves as a map, storing children under string keys.
```
Comment (Josmithr) [inline experimental/dds/tree2/src/class-tree/schemaTypes.ts:152]: Nit: I'm arguing for complete sentences in our public facing docs where possible for accessibility. So I would recommend rephrasing these as `A node which...`

File: experimental/dds/tree2/src/class-tree/tree.ts [hunk 2]
```diff
@@ -14,8 +14,11 @@ import {
 	InsertableTreeFieldFromImplicitField,
 	TreeFieldFromImplicitField,
 } from "./schemaTypes";
+
 /**
- * Channel for a Tree DDS.
+ * Channel for a Fluid Tree DDS.
+ * @remarks
+ * Allows storing and collaboratively editing schema aware hierarchial data.
```
Comment (Josmithr) [inline experimental/dds/tree2/src/class-tree/tree.ts:21]: Nit
```suggestion
 * Allows storing and collaboratively editing schema-aware hierarchial data.
```

File: experimental/dds/tree2/src/class-tree/tree.ts [hunk 3]
```diff
@@ -58,6 +61,7 @@ export interface ITree extends IChannel {
 }
 
 /**
+ * Configuration for how to schematize a tree.
```
Comment (Josmithr) [inline experimental/dds/tree2/src/class-tree/tree.ts:64]: Can we define "schematize" here and/or link to some existing definition elsewhere?

File: experimental/dds/tree2/src/core/revertible/revertible.ts [hunk 4]
```diff
@@ -12,6 +12,9 @@
 export interface Revertible {
 	/** Indicates the type of edit that produced this revertible. */
 	readonly kind: RevertibleKind;
+	/**
+	 * Information about which client created the edit.
+	 */
 	readonly origin: {
 		readonly isLocal: boolean;
```
Comment (Josmithr) [inline experimental/dds/tree2/src/core/revertible/revertible.ts:19]: Docs for this too? 😁

**Review by Josmithr — APPROVED** [review]
Thank you for the documentation! Always my favorite PRs to review 😁

File: experimental/dds/tree2/src/class-tree/schemaTypes.ts (hunk 1 above)
Comment (CraigMacomber) [reply experimental/dds/tree2/src/class-tree/schemaTypes.ts:152]:
```suggestion
	 * A node which serves as a map, storing children under string keys.
```

File: experimental/dds/tree2/src/class-tree/schemaTypes.ts [hunk 5]
```diff
@@ -131,9 +148,23 @@ export enum FieldKind {
...
+	 */
 	List,
+	/**
+	 * Node which stores a heterogenous collection of children in named fields.
+	 * @remarks
+	 * Each field gets its own schema.
+	 */
 	Object,
+	/**
+	 * Node which stores a single leaf value.
```
Comment (CraigMacomber) [inline experimental/dds/tree2/src/class-tree/schemaTypes.ts:166]:
```suggestion
	 * A node which stores a single leaf value.
```

File: experimental/dds/tree2/src/class-tree/schemaTypes.ts [hunk 6]
```diff
@@ -131,9 +148,23 @@ export enum FieldKind {
...
+	/**
+	 * Node which serves as a map, storing children under string keys.
+	 */
 	Map,
+	/**
+	 * Node which serves as a list, storing children in an ordered sequence.
+	 */
 	List,
+	/**
+	 * Node which stores a heterogenous collection of children in named fields.
```
Comment (CraigMacomber) [inline experimental/dds/tree2/src/class-tree/schemaTypes.ts:160]:
```suggestion
	 * A node which stores a heterogenous collection of children in named fields.
```

File: experimental/dds/tree2/src/class-tree/schemaTypes.ts [hunk 7]
```diff
@@ -131,9 +148,23 @@ export enum FieldKind {
  * @beta
  */
 export enum NodeKind {
+	/**
+	 * Node which serves as a map, storing children under string keys.
+	 */
 	Map,
+	/**
+	 * Node which serves as a list, storing children in an ordered sequence.
```
Comment (CraigMacomber) [inline experimental/dds/tree2/src/class-tree/schemaTypes.ts:156]:
```suggestion
	 * A node which serves as a list, storing children in an ordered sequence.
```

File: experimental/dds/tree2/src/core/revertible/revertible.ts (hunk 4 above)
Comment (CraigMacomber) [reply experimental/dds/tree2/src/core/revertible/revertible.ts:19]: API Extractor doesn't believe in such docs, so I'm guessing API-Documenter won't consume them? Should probably document it anyway.

Comment (CraigMacomber) [reply experimental/dds/tree2/src/core/revertible/revertible.ts:19]:
```suggestion
		/**
		 * Indicates if the {@link Revertible} is from the local client (true) or a remote client (false).
		 */
		readonly isLocal: boolean;
```

File: experimental/dds/tree2/src/class-tree/tree.ts (hunk 3 above)
Comment (CraigMacomber) [reply experimental/dds/tree2/src/class-tree/tree.ts:64]:
```suggestion
 * Configuration for how to {@link ITree.schematize|schematize} a tree.
```

**Pull request merged**